skyportal/handlers/api/candidate.py

```diff
@@ -677,117 +679,45 @@ def post(self):
+        description: Delete a candidate
         parameters:
           - in: path
-            name: obj_id
-            required: True
+            name: candidate_id
+            required: true
             schema:
-              type: string
-        requestBody:
-          content:
-            application/json:
-              schema:
-                type: object
-                properties:
-                  filter_ids:
-                    type: array
-                    items:
-                      type: integer
-                    description: List of associated filter IDs
-                  passing_alert_id:
-                    type: integer
-                    description: ID of associated filter that created candidate
-                    nullable: true
-                  passed_at:
-                    type: string
-                    description: Arrow-parseable datetime string indicating when alert passed filter.
-                    nullable: true
-                required:
-                  - filter_ids
+              type: integer
         responses:
           200:
             content:
               application/json:
                 schema: Success
-          400:
-            content:
-              application/json:
-                schema: Error
         """
-        data = self.get_json()
-        data["obj_id"] = obj_id
-
-        # Ensure user has access to candidate
-        if (
-            DBSession().query(Candidate).filter(Candidate.obj_id == obj_id).first()
-            is None
-        ):
-            return self.error("Invalid ID.")
-
-        passing_alert_id = data.pop("passing_alert_id", None)
-        passed_at = data.pop("passed_at", None)
-        if passed_at is not None:
-            passed_at = arrow.get(passed_at).datetime
-        try:
-            filter_ids = data.pop("filter_ids")
-        except KeyError:
-            return self.error("Missing required filter_ids parameter.")
-        user_accessible_filter_ids = [
-            filtr.id
-            for g in self.current_user.accessible_groups
-            for filtr in g.filters
-            if g.filters is not None
-        ]
-        if not all([fid in user_accessible_filter_ids for fid in filter_ids]):
-            return self.error(
-                "Insufficient permissions - you must only specify "
-                "filters that you have access to."
-            )
-
-        filters = Filter.query.filter(Filter.id.in_(filter_ids)).all()
-        if not filters:
-            return self.error("At least one valid filter ID must be provided.")
-
-        candidates = (
-            DBSession()
-            .query(Candidate)
-            .filter(Candidate.obj_id == obj_id, Candidate.filter_id.in_(filter_ids))
-            .all()
-        )
-
-        for candidate in candidates:
-            candidate.passed_at = passed_at
-            candidate.passing_alert_id = passing_alert_id
-
+        c = Candidate.query.get(candidate_id)
```

Comment:
Deletion should be handled with care and should follow the ownership model - super admins and candidate creators should be the only ones allowed to delete something. This implementation makes it possible for anyone with the manage sources ACL to delete a candidate - even if they are not in the group that a candidate belongs to !

Reply:
Done